app/models/user.rb

```diff
@@ -165,7 +160,12 @@ def create_profile
   has_many :reporter_feedback_messages, class_name: "FeedbackMessage",
                                         inverse_of: :reporter, foreign_key: :reporter_id, dependent: :nullify
   has_many :response_templates, inverse_of: :user, dependent: :destroy
-  has_many :tweets, dependent: :destroy
+  has_many :source_authored_user_subscriptions, class_name: "UserSubscription",
+                                                foreign_key: :author_id, inverse_of: :author, dependent: :destroy
+  has_many :subscribed_to_user_subscriptions, class_name: "UserSubscription",
+                                              foreign_key: :subscriber_id, inverse_of: :subscriber, dependent: :destroy
+  has_many :subscribers, through: :source_authored_user_subscriptions, dependent: :destroy
+  has_many :tweets, dependent: :nullify
```

Comment:
I switched this from destroy to nullify as Tweets are only joined to the user if we find a matching twitter screen name, but the row itself can be kept in the DB as Tweets are usually embedded in other people's articles anyway